### ISSUE TITLE
Chunks are relatively referenced even when the main js file is loaded using a custom mix URL

### ISSUE THREAD

**Issue by danhanly** [laravel-mix/laravel-mix#3205]
-   Laravel Mix Version: 6.0.39 (`npm list --depth=0`)
-   Node Version (`node -v`): v16.9.1
-   NPM Version (`npm -v`): 8.1.3
-   OS: Mac 11.6

### Description:
Chunks are relatively referenced even when the main js file is loaded using a custom mix URL.

Here's an output from devtools:

| location | resource |
|---------|----------|
| cdn.example.com | app.js |
| localhost | 163.js |
| localhost | 752.js |

Here's my config/app.php:

```
'mix_url' => env('MIX_ASSET_URL', '/'),
```

webpack.mix.js:

```
mix.js('resources/js/app.js', 'public/js')
    .copy('resources/svg', 'public/svg')
    .vue()
    .sourceMaps()
    .postCss('resources/css/app.css', 'public/css', [
        require("tailwindcss"),
    ])
```

The `mix_url` config option appears to work for the main app.js file, but all chunk files contained within are relatively referenced.

They're present on the CDN, but not on the localhost, which means they trigger a 404 error. If the chunks correctly referenced the CDN, they'd load fine.

### Steps To Reproduce:

1. Set Mix up to Automatically Chunk the files
2. Set a custom asset URL using the `mix_url` app.php config file
3. Run `npm run prod` and verify the `app.js` file correctly references the chosen `mix_url` config option
4. Verify that any numbered chunks do not follow this URL and are instead referenced locally

**Comment by danhanly** [comment]
Sidenote:

The reason this matters to me is that I'm using an Application Load Balancer to route users to different applications based upon the path, so all static assets are being deployed to a CDN to simplify the routing rules and avoid duplication between the platforms. e.g. two platforms may have a `/js/app.js` and because the routing scheme is path-based, I can't tell the requester which `/js/app.js` to download.

So CDN is the best option for dealing with all assets as I can host them in different S3 buckets/prefixes to stop my ALB path rules from requesting the wrong file.

**Comment by mike-apiabroad** [comment]
We have just run into this issue as well. Same exact thing, main .js is fine, chunks are loading relative.

@danhanly i'll let you know if we find anything

**Comment by thecrypticace** [comment]
Yep that's because Mix itself does not use the `MIX_ASSET_URL` env var but it would make sense for us to read it during build to augment chunk paths. I'm uncertain how easy that is (the path still has to be relative for the build but the manifest/chunk output has to reference the url instead). I'll look into it.

**Comment by aakarim** [comment]
You'll need to set the runtime global variable `__webpack_public_path__` to `ASSET_URL`. 

`webpack.mix.js`:
```
require('dotenv').config(); // load the config from Laravel's .env
// ...

// ensure that we are in production when building this, since HMR won't work
// with dynamic public paths.
if (process.env.ASSET_URL && mix.inProduction()) {
  console.log('using dynamic asset URL');
  mix.define({
    'process.env.ASSET_URL': JSON.stringify(process.env.ASSET_URL),
  });
}
```

create a file called `public-path.js` that needs to be loaded before anything else in your app:
```
if (process.env.ASSET_URL) {
  __webpack_public_path__ = process.env.ASSET_URL;
}
```

then in your `entrypoint` or `main.js` file:
```
import './public-path';
// you can now load dynamic imports from your CDN 
// import('./lazy-module').then(lazyModule => lazyModule.someFunc());
```

**Comment by thomaswardiii** [comment]
> You'll need to set the runtime global variable `__webpack_public_path__` to `ASSET_URL`.
> 
> create a file called `public-path.js` that needs to be loaded before anything else in your app:
> 
> ```
> if (process.env.ASSET_URL) {
>   __webpack_public_path__ = process.env.ASSET_URL;
> }
> ```
> 
> then in your `entrypoint` or `main.js` file:
> 
> ```
> import './public-path';
> // you can now load dynamic imports from your CDN 
> // import('./lazy-module').then(lazyModule => lazyModule.someFunc());
> ```
> 
> use DefinePlugin to expose the config variable in `webpack.mix.js`:
> 
> ```
> require('dotenv').config(); // load the config from Laravel's .env
> // ...
> 
> // ensure that we are in production when building this, since HMR won't work
> // with dynamic public paths.
> if (process.env.ASSET_URL && mix.inProduction()) {
>   console.log('using dynamic asset URL');
>   mix.define({
>     'process.env.ASSET_URL': JSON.stringify(process.env.ASSET_URL),
>   });
> }
> ```
> 
> Taken from https://webpack.js.org/guides/public-path/#on-the-fly

Any particular reason you didn't use `MIX_ASSET_URL` in all of this instead of the parsing of the `ASSET_URL`? Anything with `MIX_*` in the .env file gets imported automatically as `process.env.MIX_*`
